app/controllers/partners/family_requests_controller.rb

```diff
@@ -23,9 +23,12 @@ def create
         end
       end
 
-      children = current_partner.children.active.where(id: children_ids).where.not(item_needed_diaperid: [nil, 0])
+      children = current_partner.children.active
+        .where(id: children_ids)
+        .joins(:needed_items)
+        .select('children.*', :item_id)
 
-      children_grouped_by_item_id = children.group_by(&:item_needed_diaperid)
+      children_grouped_by_item_id = children.group_by { |child| child.item_id }
```

Comment:
This can just be `group_by(&:item_id)`.

Reply:
addressed a94971ff332a44efadb70d7353d3c5eaaa0f272d